ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -330,6 +335,19 @@ public void analyze(ASTNode ast, Context ctx) throws SemanticException {
     initCtx(ctx);
     init(true);
     analyzeInternal(ast);
+    postAnalyze();
+  }
+
+  /**
+   * This method is called after analyzeInternal, assuming all the semantic context is present (analyzeInternal is
+   * overridden by the subclasses of BaseSemanticAnalyzer).
+   * The reason why postAnalyze is private is that it's not supposed to be overridden.
+   */
+  private void postAnalyze() {
+    if (queryProperties == null) {
+      queryProperties = new QueryProperties();
+    }
+    queryProperties.setUsedTables(CacheTableHelper.getUniqueNames(ctx.getParsedTables()));
```

Comment:
could we move `CacheTableHelper.getUniqueNames(ctx.getParsedTables())` on a new line

Reply:
you mean refactoring it to a separate variable for better debuggability or just a line break like below:
```
    queryProperties.setUsedTables(
        CacheTableHelper.getUniqueNames(ctx.getParsedTables()));
```

Reply:
yes, please